source/fe/mapping_fe.cc

```diff
@@ -909,12 +909,82 @@ MappingFE<dim, spacedim>::transform_real_to_unit_cell(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const Point<spacedim> &                                     p) const
 {
-  Assert(false, StandardExceptions::ExcNotImplemented());
+  const std::vector<Point<spacedim>> support_points =
+    this->compute_mapping_support_points(cell);
+
+  Tensor<1, spacedim> DF[dim];
+  // [TODO]
+  // When 2nd derivatives are implemented, we'll uncomment this
+  // Tensor<1, spacedim> D2F[dim][dim];
+
+  Point<dim>      p_unit;
+  Point<spacedim> mapped_point;
+  Point<spacedim> p_minus_F;
+
+  Point<dim>     f;
+  Tensor<2, dim> df;
+
+  const double       eps        = 1.e-12 * cell->diameter();
+  const unsigned int loop_limit = 10;
+
+  unsigned int loop = 0;
+  do
+    {
+      auto mapped_point = support_points[0] * 0.0;
+      for (unsigned int i = 0; i < this->fe->n_dofs_per_cell(); ++i)
+        {
+          mapped_point += support_points[i] * this->fe->shape_value(i, p_unit);
+          auto F = this->fe->shape_grad(i, p_unit);
```

Comment:
```suggestion
          const auto F = this->fe->shape_grad(i, p_unit);
```

Reply:
I think it would be easier to read, though, if you called this variable `G_i` or `shape_grad_i` to make it clearer what it corresponds to.